solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -274,15 +275,15 @@ contract StandardPoolConverter is ConverterVersion, IConverter, ContractRegistry
      * @return number of reserve tokens
      */
     function reserveTokenCount() public view returns (uint16) {
-        return uint16(__reserveTokens.length);
+        return _reserveTokensLength;
     }
 
     /**
      * @dev returns the array of reserve tokens
      *
      * @return array of reserve tokens
      */
-    function reserveTokens() public view returns (IERC20[] memory) {
+    function reserveTokens() public view returns (IERC20[2] memory) {
```

Comment:
isn't that an interface breaking change that also means contracts can't use the function universally on all converter versions?

Reply:
Yes, it is indeed breaking onchain API (but not offchain API).
However, once all SP converters are upgraded, the problem will be only per type, not per version (i.e., you'll need to fetch the return value differently for V1 and for SP).

Reply:
@yudilevi:
Yes, but I don't think that it is used anywhere onchain outside this contract.

Reply:
it's used by others

Reply:
We can apply a fix very easily, by reverting this function to its original version (which returns a dynamic array), and simply copy the array of tokens into a dynamic array before returning it. This is less efficient, but if this function is called only from constants functions, then there's no issue.
The issue here is that we will never be able to get rid of that inefficiency, because we cannot overload this function (as functions cannot be overloaded only by distinguishing the return-value type).
So preferably, if we know of anyone who has embedded it **in their contracts**, then we might want to try and resolve it with them.
And again - if it's not used in any contract, then we may as well change this function's prototype (rather then keeping it with an inefficient internal implementation).

Reply:
We'll figure it out in the next pool type and then we can remove the backward compatible functions.
However, the bigger question here is why are we using arrays to begin with.
This converter has exactly 2 reserves, why not just use 2 values (everywhere)? Keep is stupid simple for devs.